plugins/outputs/splunk/splunk.go

```diff
@@ -0,0 +1,298 @@
+	// -----------------------------------------------------------------------------------------------------------------
+	var payload []byte
+	var err error
+	payload, err = json.Marshal(allMetrics)
+
+	if err != nil {
+		return fmt.Errorf("unable to marshal data, %s\n", err.Error())
+	}
+	log.Printf("D! Output [Splunk] %s\n", payload)
+
+	// -----------------------------------------------------------------------------------------------------------------
+	//  Send the data to Splunk
+	// -----------------------------------------------------------------------------------------------------------------
+	req, err := http.NewRequest("POST", s.SplunkUrl, bytes.NewBuffer(payload))
+	if err != nil {
+		return fmt.Errorf("unable to create http.Request \n    URL:%s\n\n", s.SplunkUrl)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", "Splunk "+s.AuthString)
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error POST-ing metrics to Splunk[%s]  Sending Data:%s\n", err, payload)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode > 209 {
+		return fmt.Errorf("received bad status code posting to %s, %d\n\n%s\n", s.SplunkUrl, resp.StatusCode, payload)
+	}
+
+	return nil
+}
+
+
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Check for metric name ending with .value and remove it, sanitize characters if desired and add separators if needed
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldName(measureName string, fieldName string, s *Splunk) string {
+	// Add any optional metric prefix to the metric name
+	name := s.Prefix + measureName
+
+	// If the fieldName!="value" add it to the metric name  
+	if s.SimpleFields || fieldName != "value" {
+		name += s.MetricSeparator + fieldName
+	}
+	
+	// Sanitize metric name by removing special characters
+	if s.ReplaceSpecials {
+		name = sanitizedChars.Replace(name)
+	}
+
+	// Sanitize metric name using regex
+	if s.UseRegex {
+		name = sanitizedRegex.ReplaceAllLiteralString(name, "-")
+	}
+
+	// Convert all _ (underscore) chartacters in final metric name to metricSeparator ("." by default)
+	if s.ConvertPaths {
+		name = pathReplacer.Replace(name)
+	}
+	return name
+}
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Parse the value depending on the data type
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldValue(v interface{}, name string, s *Splunk) (float64, error) {
+	switch p := v.(type) {
+	case bool:
+		if s.ConvertBool {
+			if p {
+				return 1, nil
+			} else {
+				return 0, nil
+			}
+		}
+	case int, int8, int16, int32, int64:
+    	return float64(v.(int64)), nil
```

Comment:
Unfortunately this will not work. As an example, in the case where `v` is an `int8`, the type assertion `v.(int64)` will panic.
Type switches have to be repetitive. One `case` for each type.

```
switch p := v.(type) {
case bool:
  ...
case int:
  return float64(p), nil
case int8:
  return float64(p), nil
case int16:
  return float64(p), nil
case int32:
  return float64(p), nil
...
}
```
Even though each `case` statement contains the same code, they have to be separate because `p` is a different type, and thus when compiled, they aren't actually the same.

Reply:
fixed.

Reply:
Fields can only be int64, uint64, float64, string, and bool.  You can only support these types, but also include a default case that logs an error.

Reply:
I've converted all numerics to float64 in the latest commit.   String and bool are also handled.  

Is this acceptable?

Reply:
Hrm. That seems like a restriction that should be removed. The type is a limitation of the storage engine. I don't think it makes sense to impose InfluxDB's limitations on the other outputs.

Reply:
I'm open to allowing additional types, but I don't want fields to be anything and everything.

Reply:
I'm confused.   This is an output plugin that's doing the conversion.    If I'm missing something I apologize, but if there really is an issue please let me know how best to resolve it in this code.

Reply:
@tmartin14 On this output, only worry about fields being one of int64, uint64, float64, string, and bool.  Anything else just log an error and ignore it.

Regarding field types in general, I don't want each output plugin to have to anticipate a huge list of possible types, and I want the types that a field can be to be well defined.  So, I don't really want the field to be an `interface{}`, I want a variant type.  I would also like at some point to return an error if the input adds other types.  I think these steps will reduce the number of bugs around type conversion, and simplify output writing.  We can expand this list of types if needed (I've been considering adding a time type for instance).

Reply:
Thanks @danielnelson.   I'm updating this back to the original types and will re-commit.